x-pack/plugins/lens/public/editor_frame_plugin/editor_frame/workspace_panel.test.tsx

```diff
@@ -274,4 +274,136 @@ Object {
+            vis: mockVisualization,
+          }}
+          visualizationState={{}}
+          datasourcePublicAPI={mockDatasource.publicAPIMock}
+          dispatch={mockDispatch}
+          ExpressionRenderer={expressionRendererMock}
+        />
+      );
+    });
+
+    it('should immediately transition if exactly one suggestion is returned', () => {
+      const expectedTable = {
+        datasourceSuggestionId: 0,
+        isMultiRow: true,
+        columns: [],
+      };
+      mockDatasource.getDatasourceSuggestionsForField.mockReturnValueOnce([
+        {
+          state: {},
+          table: expectedTable,
+        },
+      ]);
+      mockVisualization.getSuggestions.mockReturnValueOnce([
+        {
+          score: 0.5,
+          title: 'my title',
+          state: {},
+          datasourceSuggestionId: 0,
+        },
+      ]);
+
+      instance.childAt(0).prop('onDrop')({
+        name: '@timestamp',
+        type: 'date',
+        searchable: false,
+        aggregatable: false,
+      });
+
+      expect(mockDatasource.getDatasourceSuggestionsForField).toHaveBeenCalledTimes(1);
+      expect(mockVisualization.getSuggestions).toHaveBeenCalledWith(
+        expect.objectContaining({
+          tables: [expectedTable],
+        })
+      );
+      expect(mockDispatch).toHaveBeenCalledWith({
+        type: 'SWITCH_VISUALIZATION',
+        newVisualizationId: 'vis',
+        initialState: {},
+        datasourceState: {},
+      });
+    });
+
+    it('should immediately transition to the first suggestion if there are multiple', () => {
+      mockDatasource.getDatasourceSuggestionsForField.mockReturnValueOnce([
+        {
+          state: {},
+          table: {
+            datasourceSuggestionId: 0,
+            isMultiRow: true,
+            columns: [],
+          },
+        },
+        {
+          state: {},
+          table: {
+            datasourceSuggestionId: 1,
+            isMultiRow: true,
+            columns: [],
+          },
+        },
+      ]);
+      mockVisualization.getSuggestions.mockReturnValueOnce([
+        {
+          score: 0.8,
+          title: 'first suggestion',
+          state: {
+            isFirst: true,
+          },
+          datasourceSuggestionId: 1,
+        },
+        {
+          score: 0.5,
+          title: 'second suggestion',
+          state: {},
+          datasourceSuggestionId: 0,
+        },
+      ]);
+
+      instance.childAt(0).prop('onDrop')({
+        name: '@timestamp',
+        type: 'date',
+        searchable: false,
+        aggregatable: false,
+      });
+
+      expect(mockDispatch).toHaveBeenCalledWith({
+        type: 'SWITCH_VISUALIZATION',
+        newVisualizationId: 'vis',
+        initialState: {
+          isFirst: true,
+        },
+        datasourceState: {},
+      });
+    });
+
+    it("should do nothing when the visualization can't use the suggestions", () => {
+      mockDatasource.getDatasourceSuggestionsForField.mockReturnValueOnce([]);
```

Comment:
Why are you mocking the datasource suggestions to not return anything? This confused me at first, shouldn't this mock `mockVisualization.getSuggestions` instead? It is also the default return value, but IMHO it makes sense here to emphasise what's tested

Reply:
Fixed